src/pybind/mgr/mgr_module.py

```diff
@@ -426,11 +426,14 @@ def call(self,
              mgr: Any,
              cmd_dict: Dict[str, Any],
              inbuf: Optional[str] = None) -> HandleCommandResult:
-        kwargs = self._collect_args_by_argspec(cmd_dict)
-        if inbuf:
-            kwargs['inbuf'] = inbuf
-        assert self.func
-        return self.func(mgr, **kwargs)
+        try:
```

Comment:
honestly, if possible I'd rather not make a change to such a core piece of mgr module code to fix a pretty minor bug unless it's really the only way to do it. Is there any chance this could be fixed in the orchestrator? Maybe the to_format function in orchestrator.py? It's just that this change could theoretically affect any call to `call` which is done for all sorts of reasons with all sorts of arguments so for any change made here it's very difficult to judge the impact.

Reply:
Yeah, I fixed it at the mgr module because the exception is raised at that level (see following BT):

```
Error EINVAL: Traceback (most recent call last):
  File "/usr/share/ceph/mgr/mgr_module.py", line 1704, in _handle_command
    return self.handle_command(inbuf, cmd)
  File "/usr/share/ceph/mgr/orchestrator/_interface.py", line 171, in handle_command
    return dispatch[cmd['prefix']].call(self, cmd, inbuf)
  File "/usr/share/ceph/mgr/mgr_module.py", line 429, in call
    kwargs = self._collect_args_by_argspec(cmd_dict)
  File "/usr/share/ceph/mgr/mgr_module.py", line 422, in _collect_args_by_argspec
    kwargs[k] = CephArgtype.cast_to(tp, v)
  File "/lib/python3.6/site-packages/ceph_argparse.py", line 244, in cast_to
    return tp(v)
  File "/lib64/python3.6/enum.py", line 293, in __call__
    return cls.__new__(cls, value)
  File "/lib64/python3.6/enum.py", line 535, in __new__
    return cls._missing_(value)
  File "/lib64/python3.6/enum.py", line 548, in _missing_
    raise ValueError("%r is not a valid %s" % (value, cls.__name__))
ValueError: 'xml' is not a valid Format

```
Another alternative would be to add xml/pretty-xml options and then handle the logic in the from/to xml calls.

Reply:
Ok, I changed the implementation to move the logic to orchestrator. I think the new impl is more clear and less risky